arviz/data/io_pymc3.py

```diff
@@ -121,8 +121,7 @@ def arbitrary_element(dct: Dict[Any, np.ndarray]) -> np.ndarray:
     def find_observations(self) -> Optional[Dict[str, Var]]:
         """If there are observations available, return them as a dictionary."""
         has_observations = False
-        if self.trace is not None:
```

Comment:
I don't really like leaving out the check that tells the caller why they are not getting observations out of their trace when they expect it.  Perhaps replace my error with a `UserWarning`?
It seems like Arviz tries to get whatever information it can out of a trace, and ignores whatever it can't figure out.  This makes it relatively robust, but means that the user doesn't know that they could get *more* information into the `InferenceData` by providing a model.

Reply:
I completely agree, I was just waiting in order to replace it with an informative warning or a deprecation warning.